### PR TITLE
Size based eviction for server-side temp files cleanup

### PR DESCRIPTION
**Pull request by bowenliang123**
# :mag: Description
## Issue References 🔗

This pull request fixes #

## Describe Your Solution 🔧

- adding `maximumSize` to support size based eviction for server-side temp files cleanup in `TempFileService`
- size-based eviction is disabled by default , with `maximumSize` set to optional by default
- time-based eviction time is now extended from 14 days to 30 days by default

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklist 📝

- [ ] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 1]
```diff
@@ -3121,6 +3121,14 @@ object KyuubiConf {
       .timeConf
       .createWithDefaultString("P14D")
 
+  val SERVER_TEMP_FILE_EXPIRE_MAX_COUNT: ConfigEntry[Long] =
+    buildConf("kyuubi.server.tempFile.maxCount")
+      .doc("The upper threshold size of server-side temporary file paths to cleanup")
+      .version("1.10.0")
+      .serverOnly
+      .longConf
+      .createWithDefault(10000)
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:3130]: how much memory will be occupied?

Comment (bowenliang123) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:3130]: Estimated in a map struct, the total memory consumption would be approximately:
max count 10000 * (key length 13+6 + value length ~50 ) =  2,020,000B ~= 2MB

Comment (bowenliang123) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:3130]: Or do you want me to increase the default value?

Comment (pan3793) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:3130]: A potential issue, what will happen if an active file expires.

Comment (bowenliang123) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:3130]: Both time-based and size-based eviction is a fallback approach to clean up the temp files. It's assumed that the threshold is large enough for use.

Comment (pan3793) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:3130]: 14d might be long enough, 10000 is relatively small, how about disabling the size-based eviction strategy by default?

Comment (bowenliang123) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala]: OK. 1. Disabled size-based eviction strategy by default ; 2. Increased the default time-base eviction time to 30days.

**Review by pan3793 — APPROVED** [review]

**Comment by bowenliang123** [comment]
Thanks, merged to master (1.10.0).

**Pull request closed without merging**